tests/scripts/thread-cert/border_router/test_advertising_proxy.py

```diff
@@ -208,20 +208,56 @@ def test(self):
-            self.host_check_mdns_service(host, host_address, 'my-service', service_port)
+            self.check_host_and_service(server, client, [host_address], 'my-service', service_port)
+            self.host_check_mdns_service(host, [host_address], 'my-service', service_port)
 
         #
-        # 9. Check if the expired service is removed by the Advertising Proxy.
+        # 9. Check if Advertising Proxy filters out Mesh Local and Link Local host addresses
         #
+        client.srp_client_remove_host()
+        self.simulator.go(2)
+        client.srp_client_set_host_name('my-host')
+        client.srp_client_set_host_address('2001::1', '2002::2',
+                                           client.get_ip6_address(config.ADDRESS_TYPE.OMR)[0],
+                                           client.get_ip6_address(config.ADDRESS_TYPE.LINK_LOCAL),
+                                           client.get_ip6_address(config.ADDRESS_TYPE.ML_EID))
+        client.srp_client_add_service('my-service', '_ipps._tcp', 12345)
+        client.srp_client_enable_auto_start_mode()
+        self.simulator.go(10)
+        self.check_host_and_service(server, client, [
+            '2001::1', '2002::2',
+            client.get_ip6_address(config.ADDRESS_TYPE.OMR)[0],
+            client.get_ip6_address(config.ADDRESS_TYPE.LINK_LOCAL),
+            client.get_ip6_address(config.ADDRESS_TYPE.ML_EID)
+        ], 'my-service', 12345)
+        self.host_check_mdns_service(
+            host, ['2001::1', '2002::2', client.get_ip6_address(config.ADDRESS_TYPE.OMR)[0]], 'my-service', 12345)
+
+        client.srp_client_set_host_address(client.get_ip6_address(config.ADDRESS_TYPE.LINK_LOCAL),
+                                           client.get_ip6_address(config.ADDRESS_TYPE.ML_EID), client.get_rloc())
+        self.simulator.go(10)
+        self.check_host_and_service(server, client, [
+            client.get_ip6_address(config.ADDRESS_TYPE.LINK_LOCAL),
+            client.get_ip6_address(config.ADDRESS_TYPE.ML_EID),
+            client.get_rloc()
+        ], 'my-service', 12345)
+        self.host_check_mdns_service(host, [], 'my-service', 12345)
+
+        client.srp_client_set_host_address('2005::3')
+        self.simulator.go(10)
+        self.check_host_and_service(server, client, ['2005::3'], 'my-service', 12345)
+        self.host_check_mdns_service(host, ['2005::3'], 'my-service', 12345)
 
+        #
+        # 10. Check if the expired service is removed by the Advertising Proxy.
+        #
         client.srp_client_stop()
         self.simulator.go(LEASE + 2)
 
         self.assertIsNone(host.discover_mdns_service('my-service', '_ipps._tcp', 'my-host'))
         self.assertIsNone(host.discover_mdns_service('my-service-1', '_ipps._tcp', 'my-host'))
 
-    def host_check_mdns_service(self, host, host_addr, service_instance, service_port=12345):
+    def host_check_mdns_service(self, host, host_addrs, service_instance, service_port=12345):
```

Comment:
```suggestion
    def host_check_mdns_service(self, host, host_addrs:Union[str, List[str]], service_instance, service_port=12345):
```
allowing both `str` for one addr or list of addrs may reduce code change.
But it's nit.